Load car categories from db for user selection

tonia4ik-s opened the pull request:
In the modal window for adding a car, load the received categories for user selection, and accordingly create a car with the selected category.

Maryna made changes because the author of the PR could not do it alone for some reason

tonia4ik-s committed:
car category

tonia4ik-s committed:
car categories load from db

marynakernychna commented on src/api/carCategory.js:
```diff
@@ -0,0 +1,9 @@
+    {
+        return instance.get(CAR_CATEGORY_URLS.GET_CAR_CATEGORIES);
+    }
+}
```
Empty line at EoF

marynakernychna commented on src/constants/url.js:
```diff
@@ -1,7 +1,8 @@
-export const SERVER_URL = "https://localhost:44319/api";
+export const SERVER_URL = "https://localhost:5001/api";
```
We have another port for service

marynakernychna commented on src/constants/url.js:
```diff
@@ -18,3 +19,7 @@ export const CAR_URLS = {
 };
+
+export const CAR_CATEGORY_URLS = {
+    GET_CAR_CATEGORIES: CAR_CATEGORY_URL + "/categories"
```
The path does not work

marynakernychna reviewed (changes requested):
> The path to the service is incorrect, so I can't finish the review
> Categories, respectively, are not written in the drop-down list, through the first point
> Please also return the deleted file package-lock.json

tonia4ik-s committed:
package-lock.json undo

tonia4ik-s committed:
carCategories empty alert

marynakernychna commented on src/components/newCarModal/addCar.js:
```diff
@@ -41,15 +59,28 @@ class NewCarModal extends React.Component {
-                                       pattern: new RegExp("^[a-zA-Z\\d ]*$"),
+                                       pattern: new RegExp("^[A-Z]"),
+                                       message:
+                                       carErrorMessages.CANNOT_START_WITH_SPACE_MESSAGE
```
Uncorrent error message

![Screenshot 2022-05-17 153213](https://user-images.githubusercontent.com/56841893/168811678-33463925-ac1b-4505-8e1b-33edf3300446.jpg)

marynakernychna commented on src/components/newCarModal/addCar.js:
```diff
@@ -60,9 +91,10 @@ class NewCarModal extends React.Component {
                                ]}
                     >
-                        <Input addonBefore="Model:" placeholder="Model"/>
+                        <Input /*addonBefore="Model:"*/ placeholder="Model"/>
```
I think you can delete the comment

marynakernychna commented on src/services/carCategoryService.js:
```diff
@@ -0,0 +1,34 @@
+            },
+            (err) => {
+                store.dispatch(closeModal());
+                err.response.status === 404
```
I think you should show a message for any error

marynakernychna commented on src/services/carService.js:
```diff
@@ -20,16 +20,21 @@ export function addCar(values) {
-                    generalErrorMessages.SOMETHING_WENT_WRONG
-                );
+            (err) => {
+                err.response.status === 406
```
I think you should display a message for any error

tonia4ik-s committed:
error messages fixed

marynakernychna committed:
Add accidentally deleted fields to the file

> Add accidentally deleted fields to the package-lock.json

PS. The author of the PR could not do it alone for some reason

> Delete brackets

PS. I was asked, so that the author does not make a separate commit for it
